### DISCUSSION TITLE
How do I get a developer build up and running?

### DISCUSSION THREAD

**Discussion by pieterdd** [tom-james-watson/breaktimer-app#107] [Q&A] [answered]
Hi! I was toying around with SVGs so I could perhaps contribute a "dynamic tray icon" feature where the cup starts full and gradually empties until the next break approaches, allowing users to better anticipate breaks.

I was wondering though, how do I get a dev build up and running? I tried `npm run build` followed by `npm run start` but I'm getting errors about icon.png not being present in the `node_modules\electron\dist\resources\app\resources\tray` directory. I'm building on Windows 10 (PowerShell, not WSL).

**Comment by tom-james-watson** [comment, marked as answer]
That sounds like a really cool idea! Development docs are here: https://github.com/tom-james-watson/breaktimer-app/blob/master/DEVELOPMENT.md. 

Should just be `npm i` followed by `npm run dev`.

I'm actually currently doing a big upgrade of the app and potentially a lot of the code will change, but if you produce something that is nice and modular it shouldn't be a problem integrating it in.

One thing you'll want to check is whether an svg can actually be used as a tray icon on all 3 supported platforms. It seems like it may not be.

**Reply by tom-james-watson** [reply]
I wonder if there is some magic that can be done with multiple pngs though.

**Reply by pieterdd** [reply]
Thanks, will take a look again when I have time.

As for using PNGs for compatibility reasons: seems like a good idea. I suppose that generating them for 100%, 90%, 80%, ..., 10% and 0% will be a good enough approximation. I'll fire up my improvised SVG generator and see if I can make decent PNGs from it.

**Reply by pieterdd** [reply]
Some test renders:

![icon-status-100](https://user-images.githubusercontent.com/1407980/134073572-800aee2c-2222-4446-a26f-1d228bfcb688.png)
![icon-status-80](https://user-images.githubusercontent.com/1407980/134073571-825b8996-9f31-4ac4-a6d2-cf2947843365.png)
![icon-status-60](https://user-images.githubusercontent.com/1407980/134073568-dd10374e-0225-4950-894b-1960805bff99.png)
![icon-status-40](https://user-images.githubusercontent.com/1407980/134073566-7c26f923-4d49-413b-a35d-120a7fe65454.png)
![icon-status-20](https://user-images.githubusercontent.com/1407980/134073563-bbdd0808-ffb1-4011-a287-ad942c21412c.png)
![icon-status-0](https://user-images.githubusercontent.com/1407980/134073559-f15ac441-e4bf-4492-9cdb-d974857452ac.png)

**Reply by danielaj28** [reply]
Looks great

**Reply by tom-james-watson** [reply]
That looks so awesome! We'd need pngs of each one, along with ideally a source file so we can recreate them. 

For macOS the tray icon needs to be monochrome: so just either solid black or transparent. MacOS then auto recolours it depending on your theme. See the `tray-iconTemplate` files here: https://github.com/tom-james-watson/breaktimer-app/tree/master/resources/tray. So that would require a separate set of icons. We could always just not support macOS for the dynamic icon though if you don't want to make that.
